src/compiler/sys.ts

```diff
@@ -869,6 +875,38 @@ namespace ts {
             };
             return nodeSystem;
 
+            /**
+             * Uses the builtin inspector APIs to capture a CPU profile
+             * See https://nodejs.org/api/inspector.html#inspector_example_usage for details
+             */
+            function enableCPUProfiler(path: string, cb: () => void) {
+                const inspector: typeof import('inspector') = require('inspector');
+                const session = new inspector.Session();
+                session.connect();
+
+                session.post('Profiler.enable', () => {
+                    session.post('Profiler.start', () => {

```

Comment:
Does this somehow capture the other configuration options?  E.g. will we know whether or not it was an incremental build?

Reply:
Nope. This _just_ captures a CPU profile. Any other diagnostic data has to be sourced elsewhere (eg, from a second invocation with `--showConfig`)

Reply:
Although the stacks in the cpu profile should make it preeeeety obvious if the trace is from a `-b`, or a `--incremental`.